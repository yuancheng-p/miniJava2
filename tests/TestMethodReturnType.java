class I {}
class A extends I{
  int a1 = 1;
  String a6 = "jsdjkndjk\njdsdjkf";
}

abstract class B {
  int a5 = 1;
  boolean a2 = false;
  public int fooA(){
    int i = 1;
    int y = 2;
    return 1+1;
  }

  public A fooA1(){
    int i = 1;
    int y = 2;
    return new A();
  }

  public I fooA2(){
    int i = 1;
    int y = 2;
    return new A();
  }

  public int foo () {
    int i = 0;
    if (i > 0)
      return 1;
    else
      return 0;
  }

  public int fo () {
    return 1;
  }
  public void fo1 () {
    return;
  }

  // fooB will not pass
  public void fooB(){
    int i = 1;
    int y = 0;
    return 1+1;
  }

  // fooB1 will not pass
  public A fooB1(){
  int i = 1;
  int y = 0;
  return new I();
  }
}
