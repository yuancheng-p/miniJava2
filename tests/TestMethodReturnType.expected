successfull parsing
start typing
check add and methods:B:fooA
check add and methods:B:fooA1
check add and methods:B:fooA2
check add and methods:B:foo
check add and methods:B:fo
check add and methods:B:fo1
check add and methods:B:fooB
check add and methods:B:fooB1
--------- env --------
A : class_env(parent=: I)
 attr:  String a6 = "jsdjkndjk\njdsdjkf";
 attr:  int a1 = 1;
B : class_env(parent=: Object)
method : fooB
method : fooA
method : foo
method : fooB1
method : fo1
method : fooA2
method : fooA1
method : fo
 attr:  boolean a2 = false;
 attr:  int a5 = 1;
Boolean : class_env(parent=: Object)
I : class_env(parent=: Object)
Integer : class_env(parent=: Object)
Object : class_env(parent=: )
String : class_env(parent=: Object)
